### PR TITLE
expression: implement vectorized evaluation for `builtinWeekDaySig`

### PR DESCRIPTION
**Pull request by chacha923**
What problem does this PR solve?
Implement vectorized evaluation for builtinWeekDaySig for #12103.

What is changed and how it works?
```
BenchmarkVectorizedBuiltinTimeFunc/builtinWeekDaySig-VecBuiltinFunc-8                                 	    8120	    125199 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinWeekDaySig-NonVecBuiltinFunc-8                              	    6962	    152873 ns/op	       0 B/op	       0 allocs/op
```
Check List
Tests

Unit test

### CODE REVIEW

File: expression/builtin_time_vec.go [hunk 1]
```diff
@@ -181,11 +181,42 @@ func (b *builtinDayNameSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
...
+	result.ResizeInt64(n, false)
+	result.MergeNulls(buf)
+	i64s := result.Int64s()
+	ds := buf.Times()
+	for i := 0; i < input.NumRows(); i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		if ds[i].IsZero() {
+			if b.ctx.GetSessionVars().SQLMode.HasNoZeroDateMode() {
```
Comment (qw4990) [inline expression/builtin_time_vec.go]: It seems we don't need to check the `NoZeroDateMode` as the old implementation.

Comment (chacha923) [reply expression/builtin_time_vec.go]: ok, i have removed NoZeroDateMode check

File: expression/builtin_time_vec.go [hunk 2]
```diff
@@ -183,11 +183,35 @@ func (b *builtinDayNameSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
...
+	result.MergeNulls(buf)
+	i64s := result.Int64s()
+	ds := buf.Times()
+	for i := 0; i < input.NumRows(); i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		if ds[i].IsZero() {
+			i64s[i] = 0
+			continue
```
Comment (qw4990) [inline expression/builtin_time_vec.go]: Should we set this row to null and use `handleInvlaidTimeError` to handle this error in this case?

File: expression/builtin_time_vec.go [hunk 3]
```diff
@@ -183,11 +183,35 @@ func (b *builtinDayNameSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
...
 
 func (b *builtinWeekDaySig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDatetime, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalTime(b.ctx, input, buf); err != nil {
```
Comment (XuHuaiyu) [inline expression/builtin_time_vec.go]: s / `err :=` / `err =`

Comment (chacha923) [reply expression/builtin_time_vec.go]: fix it

File: expression/builtin_time_vec.go [hunk 4]
```diff
@@ -183,11 +183,38 @@ func (b *builtinDayNameSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
...
 func (b *builtinWeekDaySig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDatetime, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err = b.args[0].VecEvalTime(b.ctx, input, buf); err != nil {
+		return err
```
Comment (XuHuaiyu) [inline expression/builtin_time_vec.go]: should we call `handleInvalidTimeError` here?

Comment (qw4990) [reply expression/builtin_time_vec.go]: This error returned by its child is not an `InvalidTimeError`, so we should return it directly here.

File: expression/builtin_time_vec.go [hunk 5]
```diff
@@ -183,11 +183,38 @@ func (b *builtinDayNameSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
...
+	result.ResizeInt64(n, false)
+	result.MergeNulls(buf)
+	i64s := result.Int64s()
+	ds := buf.Times()
+	for i := 0; i < input.NumRows(); i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		if ds[i].IsZero() {
+			if err = handleInvalidTimeError(b.ctx, types.ErrIncorrectDatetimeValue.GenWithStackByArgs(ds[i].String())); err != nil {
```
Comment (SunRunAway) [inline expression/builtin_time_vec.go]: Is the `err != nil` judgement is redundant here?

Comment (qw4990) [reply expression/builtin_time_vec.go]: `handleInvalidTimeError` may return `nil` in some cases.

**Comment by qw4990** [comment]
@chacha923 Please sign the CLA license.

**Pull request merged**